internal/printer/printer.go

```diff
@@ -238,6 +274,135 @@ func BoldRed(msg interface{}) string {
+		return BoldRed("> 3. Failed switching to primary"), false
+	case ps.DataImportDetachExternalDatabaseRunning:
+		return BoldYellow("> 4. detaching external database"), true
+	case ps.DataImportDetachExternalDatabaseError:
+		return BoldRed("> 4. failed to detach external database"), false
+	case ps.DataImportReady:
+		return BoldGreen("> 5. Ready"), false
+	}
+
+	panic("unhandled state " + d.String())
+}
+
+func GetPendingImportStates(state ps.DataImportState) string {
+	var pendingStates []string
+	switch state {
+	case ps.DataImportPreparingDataCopy:
+		pendingStates = append(pendingStates, BoldBlack("2. Copied Data"))
+		pendingStates = append(pendingStates, BoldBlack("3. Running as Replica"))
+		pendingStates = append(pendingStates, BoldBlack("4. Running as Primary"))
+		pendingStates = append(pendingStates, BoldBlack("5. Detached external database"))
+	case ps.DataImportCopyingData:
+		pendingStates = append(pendingStates, BoldBlack("3. Running as Replica"))
+		pendingStates = append(pendingStates, BoldBlack("4. Running as Primary"))
+		pendingStates = append(pendingStates, BoldBlack("5. Detached external database"))
+	case ps.DataImportSwitchTrafficPending:
+		pendingStates = append(pendingStates, BoldBlack("4. Running as Primary"))
+		pendingStates = append(pendingStates, BoldBlack("5. Detached external database"))
+	case ps.DataImportSwitchTrafficCompleted:
+		pendingStates = append(pendingStates, BoldBlack("5. Detached external database"))
+	}
+	return strings.Join(pendingStates, "\n")
+}
+
+func ImportProgress(state ps.DataImportState) string {
+	preparingDataCopyState := color.New(color.FgBlack).Add(color.Bold).Sprint("1. Started Data Copy")
+	dataCopyState := color.New(color.FgBlack).Add(color.Bold).Sprint("2. Copied Data")
+	switchingTrafficState := color.New(color.FgBlack).Add(color.Bold).Sprint("3. Running as Replica")
+	detachExternalDatabaseState := color.New(color.FgBlack).Add(color.Bold).Sprint("4. Running as Primary")
+	readyState := color.New(color.FgBlack).Add(color.Bold).Sprint("5. Detached External Database, ready to use")
+	// Preparing data copy > Data copying > Running as Replica > Running as Primary > Detach External database
+	switch state {
+	case ps.DataImportPreparingDataCopy:
+		preparingDataCopyState = color.New(color.FgYellow).Add(color.Bold).Sprint(state.String())
+	case ps.DataImportSwitchTrafficPending:
+		preparingDataCopyState = color.New(color.FgGreen).Add(color.Bold).Sprint("1. Started Data Copy")
+		dataCopyState = color.New(color.FgGreen).Add(color.Bold).Sprint("2. Copied Data")
+		switchingTrafficState = color.New(color.FgYellow).Add(color.Bold).Sprint("3. Running as Replica")
+	case ps.DataImportPreparingDataCopyFailed:
+		preparingDataCopyState = color.New(color.FgGreen).Add(color.Bold).Sprint("1. Cannot start data copy")
+	}
+
+	return strings.Join([]string{preparingDataCopyState, dataCopyState, switchingTrafficState, detachExternalDatabaseState, readyState}, "\n")
+
+}
```

Comment:
All the import related printer functions should move to `internal/cmd/dataimports`. The `printer` package only contains generic printing utilities.